numpy/lib/function_base.py

```diff
@@ -514,7 +514,10 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
+        # Cast bool, unsigned int, and int to float64 by default
```

Comment:
I agree that before <code>a = a.astype('float64')</code> using <code>dt = np.result_type(a, 0.0)</code> would be appropriate. However, as @jaimefrio  mentioned before <code>astype</code>  requires an intermediate array for the conversion. So I was thinking of using <code>a = np.multiply(1.0, a)</code>.
So the fix would look like the following:
```
if issubclass(a.dtype.type, (integer, bool)):
    a = np.multiply(1.0, a)
```
Otherwise I could do it the way using <code>result_type</code> then the fix would look like this:
```
if issubclass(a.dtype.type, (integer, bool)):
    dt = np.result_type(a, 0.0)
    a = a.astype(dt)
```